packages/bigquery/src/index.js

```diff
@@ -57,16 +57,30 @@ var Table = require('./table.js');
  *
  * @resource [What is BigQuery?]{@link https://cloud.google.com/bigquery/what-is-bigquery}
  *
+ * The apiEndpoint from options will set the host. If not set, the
+ * `GOOGLE_CLOUD_BIGQUERY_ENDPOINT` environment variable is honored,
+ * otherwise the actual API endpoint will be used.
+ *
  * @param {object} options - [Configuration object](#/docs).
+ * @param {object} options.apiEndpoint - Override the default API endpoint
+ * to reach BigQuery.
  */
 function BigQuery(options) {
   if (!(this instanceof BigQuery)) {
     options = common.util.normalizeArguments(this, options);
     return new BigQuery(options);
   }
 
+  var basePath = '/bigquery/v2';
+
+  var baseInfo = common.util.determineBaseUrl(
+      options,
+      'GOOGLE_CLOUD_BIGQUERY_ENDPOINT',
+      'https://www.googleapis.com');
+
   var config = {
-    baseUrl: 'https://www.googleapis.com/bigquery/v2',
+    baseUrl: baseInfo.apiEndpoint + basePath,
+    customEndpoint: baseInfo.customEndpoint,
```

Comment:
Actually, for the protoServices, what if we piggy back on the fact that we can tell if the endpoint is custom or not. If it's custom we override every baseUrl otherwise we leave them as they are. As for passing an array I'm fine with that

Slight tangent, googlebot is still complaining about the CLA, is there any way for you to check that  rodrigo@twosigma.com is part of the allowed email addresses to commit code from Two Sigma?